### PR TITLE
feat(pci.storage): add OMK encryption for S3 storage

### PR DESCRIPTION
**Pull request by darsene**
ref: MANAGER-12924

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/kms-omk`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-12924,MANAGER-12954
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

**Review by dectotam — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/storages/object-storage/objects/add/add.routing.js [hunk 1]
```diff
@@ -36,6 +36,21 @@ export default /* @ngInject */ ($stateProvider) => {
...
+      encryptionAlgorithms: /* @ngInject */ ($http) =>
+        $http
+          .get('/cloud.json')
+          .then(
+            ({ data: { models } }) =>
+              models['cloud.storage.EncryptionAlgorithmEnum']?.enum || [],
+          ),
+      encryptionAvailable: /* @ngInject */ (ovhFeatureFlipping) =>
+        ovhFeatureFlipping
+          .checkFeatureAvailability('key-management-system:storage-encryption')
```
[the remark this answers is not in the document]
Comment (anooparveti) [reply packages/manager/modules/pci/src/projects/project/storages/object-storage/objects/add/add.routing.js:48]: I would say `object-storage:encryption`

[the remark this answers is not in the document]
Comment (darsene) [reply packages/manager/modules/pci/src/projects/project/storages/object-storage/objects/add/add.routing.js]: fixed

File: packages/manager/modules/pci/src/projects/project/storages/object-storage/objects/add/add.routing.js [hunk 2]
```diff
@@ -36,6 +36,23 @@ export default /* @ngInject */ ($stateProvider) => {
...
       goBack: /* @ngInject */ (goToStorageContainers) => goToStorageContainers,
+      encryptionAlgorithms: /* @ngInject */ ($http, encryptionAvailable) =>
+        encryptionAvailable
+          ? $http
+              .get('/cloud.json')
+              .then(
+                ({ data: { models } }) =>
+                  models['cloud.storage.EncryptionAlgorithmEnum']?.enum || [],
+              )
+          : [],
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/projects/project/storages/object-storage/objects/add/add.routing.js:47]: IMO, this should not fail silently
By returning an empty array, the only thing the user will be displayed is just an introduction text with nothing to chose and no message to warn him that something went wrong

Comment (darsene) [reply packages/manager/modules/pci/src/projects/project/storages/object-storage/objects/add/add.routing.js]: fixed.

I added fallback values to ensure the availability of the service

**Review by y4nnL — APPROVED** [review]

**Pull request merged**